src/Shared/ConsoleLogs/TimestampParser.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Aspire.Dashboard.ConsoleLogs;
+
+internal static partial class TimestampParser
+{
+    private static readonly Regex s_rfc3339RegEx = GenerateRfc3339RegEx();
+
+    public static bool TryParseConsoleTimestamp(string text, [NotNullWhen(true)] out TimestampParserResult? result)
+    {
+        var match = s_rfc3339RegEx.Match(text);
+
+        if (match.Success)
+        {
+            var span = text.AsSpan();
+            var timestamp = span[match.Index..(match.Index + match.Length)];
```

Comment:
I think this can just be:

```suggestion
            var timestamp = match.ValueSpan;
```